### PR TITLE
Feature/add integration tests and api fixes

### LINKED ISSUE #25: Add integration tests for api endpoints with docker and bash scripting (check minimalpy: https://github.com/blankdots/minimalpy/tree/master/deploy/test)

### LINKED ISSUE #49: Fix api responses to be more human friendly
#### Proposed solution
Stuff that needs to be fixed:
- date result (now just python datetime dump)
- answers from /submit -endpoint should contain all generated accessionIds
- schemas should have title, detail and url in response json
- remove internal _id from api responses
- add createdDate and updatedDate

Parsing-process related stuff that needs to be fixed:
- acstract, title, descriptions have newline characters and <>-characters. 
- attribute-keys. Should be possible to remove attribute-keys and elevate values behind keys one level higher.
- children-keys. Conversion process now creates children-key and list of children. Maybe try to elevate these one step higher also? 

#### DoD (Definition of Done)
- Api responses look more human friendly

#### Testing
- Unit testing

### PR DESCRIPTION
**Pull request by otahontas**
### Description

Fixing bunch of stuff related to new api docs, adding also docker-compose(s) and initial version for integration tests.

### Related issues
Partially fixes #49 #52 

Fixes #25 

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue), refactoring

### Changes Made
- dateCreated and dateModified is added for all objects
- date results are parsed for response jsons 
- /submit endpoint response contains accession ids for all files sent through it
- /schema endpoits returns answer that corresponds to api docs (title, details, url)
- mongodb _id is removed from api responses
- integration tests for  POST/GET with single objects is added. Some queries are also tested. tests are ran against production version dockerfile asyncronously, so if wanted, one can try to to submit big amount of objects to stress test endpoints.
- new development docker-compose and dockerfile is added, allows quick development setup
- refactored some functionalities:
   - parser module is now purely for xml-json parsing. Previosly it also added accessionIds and publishDates, but its more logical to handle these in operators (in future accessionIds are needed to be added to json submissions too).
   - handlers module had some copypaste stuff between /submit and /object/{schema} -endpoints, those are moved to operators module.

Also some other small fixes, see git history for more details.

### Testing

- [x] Unit Tests
- [x] Integration Tests

### CODE REVIEW

**Comment by otahontas** [comment]
@blankdots @csc-jm PR is almost ok. Frontend master doesn't have url routing, so integration tests were failing. Had to change them to run against development version of backend (which doesn't include frontend build). 

However, now there's a new bug with aiohttp, not sure what. Would appriciate if someone would help debugging this:
- https://github.com/CSCfi/metadata-submitter/blob/feature/add-integration-tests-and-db-fixes/tests/integration/run_tests.py
- Here's the error: https://travis-ci.com/github/CSCfi/metadata-submitter/jobs/353108419

**Comment by blankdots** [comment]
@otahontas sometimes `sleep` helps :D or waiting for the services to start.

**Comment by blankdots** [comment]
getting annoyed with travis ... maybe we should switch to github actions

**Comment by otahontas** [comment]
> @otahontas sometimes `sleep` helps :D or waiting for the services to start.

Classic travis! Thanks for debugging

File: README.md [hunk 1]
```diff
@@ -4,26 +4,27 @@
...
 
 ## Install and run
 
 Requirements:
 - Python 3.6+
 - Mongodb
+- Docker + docker-compose
 
-Clone project and install it by running: `pip install .`
+For quick testing, go to the `development` folde and launch server and database with Docker by running `docker-compose up  --build` (add `-d` flag to run containers in background). Server can then be found from `http://localhost:5430`.
```
Comment (blankdots) [inline README.md:16]:
```suggestion
For quick testing, go to the `development` folder and launch server and database with Docker by running `docker-compose up  --build` (add `-d` flag to run containers in background). Server can then be found from `http://localhost:5430`.
```

File: README.md [hunk 2]
```diff
@@ -4,26 +4,27 @@
...
-
-After installing and setting up database, server can be launched with `metadata_submitter`.
+For more detailed setup, do following:
+- Install project by running: `pip install .` in project root
+- Setup mongodb and env variables via desired way, details:
+  - Server expects to find mongodb instance running, spesified with following environmental variables:
+    - `MONGO_INITDB_ROOT_USERNAME`, username for admin user to mondogb instance
+    - `MONGO_INITDB_ROOT_PASSWORD`, password for admin user to mondogb instance
+    - `MONGODB_HOST`, host and port for mongodb instance (e.g. `localhost:27017`)
+  - As a default, server uses default values from MongoDB Docker image.
```
Comment (blankdots) [inline README.md:25]:
```suggestion
  - out of the box, the `Metadata Submitter` is configured server uses default values from MongoDB Docker image.
```

File: metadata_backend/api/operators.py [hunk 3]
```diff
@@ -128,10 +208,33 @@ def __init__(self) -> None:
...
         :param data_raw: Data from mongodb query with single result.
         :returns: XML content
         """
         return data_raw["content"]
+
+    def _handle_data_and_add_to_db(self, type: str, data: str,
+                                   accession_id: str) -> None:
+        """Format added xml metadata object and add it to db.
+
+        XML is validated parsed to json with parser-class and json is added
```
Comment (blankdots) [inline metadata_backend/api/operators.py:224]:
```suggestion
        XML is validated then parsed to JSON with parser-class and is added
```

File: tests/test_handlers.py [hunk 4]
```diff
@@ -71,14 +76,13 @@ async def setUpAsync(self):
...
         self.MockedXMLOperator = self.patch_xmloperator.start()
-        self.patch_accession.start()
 
     async def tearDownAsync(self):
         """Cleanup mocked stuff."""
         self.patch_parser.stop()
         self.patch_operator.stop()
         self.patch_xmloperator.stop()
-        self.patch_accession.stop()
+        # self.patch_accession.stop() TODO: move this to operators
```
Comment (blankdots) [inline tests/test_handlers.py:85]: maybe make an issue ouf of these `TO DO` :)

**Comment by otahontas** [comment]
Fixed typos and added some tests, so coverage should be now ok again. @blankdots can you check if this is ok to merge

**Pull request merged**